File: 0006-zigzag-conversion/0006-zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if(numRows==1)
            return s;
        vector<string> mat(numRows);
        int j=0;
        bool low=true;
        for(int i=0;i<s.size();i++){
            mat[j].push_back(s[i]);
            if(low){
                if(j==numRows-1){
                    j--;
                    low=false;
                }
                else{
                    j++;
                }
            }
            else{
                if(j==0){
                    j++;
                    low=true;
                }
                else{
                    j--;
                }
            }
    
        }
        string ans="";
        for(auto s:mat){
            ans+=s;
        }
        return ans;
    }
};

File: 0006-zigzag-conversion/0006-zigzag-conversion.php
<?php

    class Solution {

    /**
     * @param String $s
     * @param Integer $numRows
     * @return String
     */
    function convert($s, $numRows) {
    if ($numRows == 1 || $numRows >= strlen($s)) {
        return $s;
    }

    $rows = array_fill(0, $numRows, '');
    $idx = 0;
    $step = 1;

    for ($i = 0; $i < strlen($s); $i++) {
        $rows[$idx] .= $s[$i];
        if ($idx == 0) {
            $step = 1;
        } else if ($idx == $numRows - 1) {
            $step = -1;
        }
        $idx += $step;
    }

    return implode('', $rows);
}

}

?>